Tasks/JenkinsQueueJob/jobsearch.ts

```diff
@@ -119,7 +119,11 @@ export class JobSearch {
             for (var i in causes) {
                 var job = thisSearch.queue.findJob(causes[i].upstreamUrl, causes[i].upstreamBuild);
                 if (job) { // we know about it
-                    if (job.state == JobState.Streaming || job.state == JobState.Finishing || job.state == JobState.Done) {
+                    if (job.state == JobState.Streaming ||
+                        job.state == JobState.Finishing ||
+                        job.state == JobState.Downloading ||
+                        job.state == JobState.Queued ||
```

Comment:
Ah, I guess that's inconsequential.   We are in this code block, that means there is a pipelined job, I don't care how it get started.  The important thing is the parent job could be in the "Queued" state, that still feels wrong to me, but that's probably somewhere else another state transition that I don't agree with.

To me, I feel if a job is "Queued" and we don't wait for its completion, we should immediately return success from this task and be done.  There shouldn't be any pipelined jobs in this case.